src/gateway/ws.rs

```diff
@@ -112,65 +175,63 @@ impl WsClient {
     ) -> Result<()> {
         debug!("[Shard {:?}] Requesting member chunks", shard_info);
 
-        let mut payload = json!({
-            "op": OpCode::GetGuildMembers.num(),
-            "d": {
-                "guild_id": guild_id.as_ref().0.to_string(),
-                "limit": limit.unwrap_or(0),
-                "nonce": nonce.unwrap_or(""),
-            },
-        });
-
-        match filter {
-            ChunkGuildFilter::None => payload["d"]["query"] = json!(""),
-            ChunkGuildFilter::Query(query) => payload["d"]["query"] = json!(query),
-            ChunkGuildFilter::UserIds(user_ids) => {
-                let ids = user_ids.iter().map(|x| x.0).collect::<Vec<u64>>();
-                payload["d"]["user_ids"] = json!(ids);
-            },
+        let (query, user_ids) = match filter {
+            ChunkGuildFilter::None => (None, None),
```

Comment:
Discord docs state that either `query` or `user_ids` is required, so this should return an empty string for the query instead